base/reducedim.jl

```diff
@@ -20,7 +20,7 @@ reduced_indices0(a::AbstractArray, region) = reduced_indices0(axes(a), region)
 function reduced_indices(inds::Indices{N}, d::Int) where N
     d < 1 && throw(ArgumentError("dimension must be ≥ 1, got $d"))
     if d == 1
-        return (reduced_index(inds[1]), tail(inds)...)
+        return (reduced_index(inds[1]), tail(inds)...)::typeof(inds)
```

Comment:
Should we backport this without these assertions?  That is, I think this patch is good for master (and perhaps even back porting to 1.4) but I don't think we should _add_ assertions in a back port — especially not for a backport to 1.0.